code/obj/item/kendo.dm

```diff
@@ -0,0 +1,245 @@
+
+    w_class = 4.0
+    two_handed = 1
+    throwforce = 4.0
+    throw_range = 4
+    stamina_crit_chance = 2
+
+    //these combat variables will change depending on the guard
+    force = 6.0
+    stamina_damage = 10
+    stamina_cost = 5.0
+
+    hit_type = DAMAGE_BLUNT
+    flags = FPRINT | TABLEPASS | USEDELAY
+    c_flags = EQUIPPED_WHILE_HELD
+    item_function_flags = USE_INTENT_SWITCH_TRIGGER | USE_SPECIALS_ON_ALL_INTENTS
+
+    var/guard
+
+    proc/change_guard(var/mob/user,var/intent)
+        user.do_disorient(10,0,0,0,0,0,null)
+        guard = intent
+        switch(guard)
+            if("help")
+                force = 5
+                stamina_damage = 10
+                stamina_cost = 5
+                item_state = "shinai-light"
+                src.setItemSpecial(/datum/item_special/simple/kendo_light)
+            if("disarm")
+                force = 6
+                stamina_damage = 10
+                stamina_cost = 8
+                item_state = "shinai-sweep"
+                src.setItemSpecial(/datum/item_special/swipe/kendo_sweep)
+            if("grab")
+                force = 6
+                stamina_damage = 15
+                stamina_cost = 10
+                item_state = "shinai-thrust"
+                src.setItemSpecial(/datum/item_special/rangestab/kendo_thrust)
+            if("harm")
+                force = 8
+                stamina_damage = 30
+                stamina_cost = 35
+                item_state = "shinai-heavy"
+                item_state = "shinai-heavy"
+                src.setItemSpecial(/datum/item_special/simple/kendo_heavy)
+        user.update_inhands()
+        src.buildTooltipContent()
+
+    proc/parry_block_check(var/mob/living/carbon/human/attacker,var/mob/living/carbon/human/defender)
+        if(attacker == defender)
+            return
+
+        if((attacker.a_intent == defender.a_intent) && !defender.hasStatus("disorient"))
+            playsound(defender, "sound/impact_sounds/kendo_parry_[pick(1,2,3)].ogg", 50, 1)
+            attacker.do_disorient(0,0,0,0,10,1)
+            return 1
+
+        else if(defender.hasStatus("blocking"))
+            playsound(defender, "sound/impact_sounds/kendo_block_[pick(1,2)].ogg", 50, 1)
+            if(attacker.equipped())
+                defender.do_disorient((attacker.equipped().stamina_cost*1.5),0,0,0,0,1,null)
+            return 2
+        return 0
+
+    proc/stat_reset()
+        if(force != 5)
+            force = 5
+        else
+            return
+        stamina_damage = 10
+        stamina_cost = 5
+        item_state = "shinai-light"
+        src.setItemSpecial(/datum/item_special/simple/kendo_light)
+        src.buildTooltipContent()
+
+    intent_switch_trigger(mob/user as mob)
+        if(guard != user.a_intent)
+            change_guard(user,user.a_intent)
+
+    attack(mob/living/carbon/human/defender as mob, mob/living/carbon/human/attacker as mob)
+        if(ishuman(defender))
```

Comment:
Working my head around this one, is this `ishuman` check needed? Theoretically, some of the bigger critters should be able to wield these (though I'm not sure how well this would work with`twohanded` right now).

Reply:
This is checking if the defender is a human, not the attacker. I'm not entirely sure if attack() is called on critters though, so the check is there just in case any strange things happen when you smack a critter.